From fe0ce958ab474164d434be572187b65ab788215b Mon Sep 17 00:00:00 2001
From: Anas Nashif <anas.nashif@intel.com>
Date: Fri, 25 Sep 2026 02:16:05 -0700
Subject: [PATCH 1/2] checkpatch: add option for excluding directories

when importing code from external sources, we do not want to fail
on checkpatch and want to run full sanity checks.

Using --exclude <dir> we should be able to exclude a list of well
defined locations in the tree that carry sources from other projects
with other styles.

Change-Id: I7d321e85eed6bc37d5c6879ae88e21d20028a433
Signed-off-by: Anas Nashif <anas.nashif@intel.com>
Signed-off-by: florent pirou <florent.pirou@intel.com>
---
 checkpatch.pl | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/checkpatch.pl b/checkpatch.pl
index 8a7787d..74bf1bb 100755
--- a/checkpatch.pl
+++ b/checkpatch.pl
@@ -55,6 +55,7 @@ my %use_type = ();
 my @use = ();
 my %ignore_type = ();
 my @ignore = ();
+my @exclude = ();
 my $help = 0;
 my $configuration_file = ".checkpatch.conf";
 my $def_configuration_dirs_help = '.:$HOME:.scripts';
@@ -113,6 +114,7 @@ Options:
   --list-types               list the possible message types
   --types TYPE(,TYPE2...)    show only these comma separated message types
   --ignore TYPE(,TYPE2...)   ignore various comma separated message types
+  --exclude DIR(,DIR22...)   exclude directories
   --show-types               show the specific message type in the output
   --max-line-length=n        set the maximum line length, (default $max_line_length)
                              if exceeded, warn on patches
@@ -339,6 +341,7 @@ GetOptions(
 	'subjective!'	=> \$check,
 	'strict!'	=> \$check,
 	'ignore=s'	=> \@ignore,
+	'exclude=s'	=> \@exclude,
 	'types=s'	=> \@use,
 	'show-types!'	=> \$show_types,
 	'list-types!'	=> \$list_types,
@@ -2924,6 +2927,16 @@ sub process {
 			$found_file = 1;
 		}
 
+		my $skipme = 0;
+		foreach (@exclude) {
+			if ($realfile =~ m@^(?:$_/)@) {
+				$skipme = 1;
+			}
+		}
+		if ($skipme) {
+			next;
+		}
+
 #make up the handle for any error we report on this line
 		if ($showfile) {
 			$prefix = "$realfile:$realline: "
-- 
2.43.0

